odxtools/structures.py

```diff
@@ -507,7 +507,8 @@ def __init__(self, *, response_type: str, **kwargs):  # "POS-RESPONSE" or "NEG-R
         self.response_type = response_type
 
     def encode(self, coded_request: Optional[ByteString] = None, **params) -> ByteString:
-        logger.info(f"Compose response message to the request {coded_request}")
+        if coded_request is not None:
+            logger.info(f"Compose response message to the request {coded_request.hex()}")
         if coded_request is not None:
```

Comment:
Duplicate `if coded_request is not None:` check

Reply:
oops. right.

Reply:
fixed, thanks!